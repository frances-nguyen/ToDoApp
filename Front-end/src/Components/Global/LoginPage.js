import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Authentication from '../AuthenticationService/Authentication'

class LoginPage extends Component {
  constructor(props) {
    super(props)
    this.state = {
      username: '',
      password: '',
      loginFailed: false
    }
    this.handleLoginChange = this.handleLoginChange.bind(this)
    this.loginClicked = this.loginClicked.bind(this)
  }

  handleLoginChange(event) {
    this.setState({
      [event.target.name]: event.target.value
    })
  }

  loginClicked() {
    if (this.state.username === 'Frances' && this.state.password === '123') {
      Authentication.registerSuccessfulLogin(
        this.state.username,
        this.state.password
      )
      this.props.history.push(`/welcome/${this.state.username}`)
    } else {
      this.setState({ loginFailed: true })
    }
  }

  render() {
    return (
      <div>
        <h1>Login</h1>
        <div className="container">
          {this.state.loginFailed && (<div className="alert alert-warning">Inavlid Credentials</div>)}
          Username:{' '}
          <input
            type="text"
            name="username"
            value={this.state.username}
            onChange={this.handleLoginChange} />
          Password:{' '}
          <input
            type="text"
            name="password"
            value={this.state.password}
            onChange={this.handleLoginChange} />
          <button className="btn btn-success" onClick={this.loginClicked}>Login</button>
        </div>
      </div>
    )
  }
}

LoginPage.propTypes = {
  history: PropTypes.object
}

export default LoginPage
