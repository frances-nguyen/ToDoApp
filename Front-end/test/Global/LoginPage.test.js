import React from 'react'
import LoginPage from '../../src/Components/Global/LoginPage'
import renderer from 'react-test-renderer'

describe('<Login /> rendering', () => {
  it('renders correctly', () => {
    let tree = renderer.create(<LoginPage />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('<Login /> interactions', () => {
  it('should change the state of username', () => {
    const event = { target: { name: 'username', value: 'username' } }
    let wrapper = shallow(<LoginPage />)
    wrapper
      .find('input')
      .first()
      .simulate('change', event)
    expect(wrapper.state('username')).toEqual('username')
    expect(wrapper.state('password')).toEqual('')
  })

  it('should change the state of password', () => {
    const event = { target: { name: 'password', value: 'password' } }
    let wrapper = shallow(<LoginPage />)
    wrapper
      .find('input')
      .first()
      .simulate('change', event)
    expect(wrapper.state('password')).toEqual('password')
    expect(wrapper.state('username')).toEqual('')
  })

  it('should call handleLoginChange', () => {
    const event = { target: { name: 'username', value: 'username' } }
    const handleLoginChange = jest.spyOn(
      LoginPage.prototype,
      'handleLoginChange'
    )
    let wrapper = shallow(<LoginPage />)
    wrapper
      .find('input')
      .first()
      .simulate('change', event)
    expect(wrapper.state('username')).toEqual('username')
    expect(handleLoginChange).toHaveBeenCalled()
  })
})
