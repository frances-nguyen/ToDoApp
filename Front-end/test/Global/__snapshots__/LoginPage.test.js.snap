// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> rendering renders correctly 1`] = `
<div>
  Username:
   
  <input
    name="username"
    onChange={[Function]}
    type="text"
    value=""
  />
  Password:
   
  <input
    name="password"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
